elyra/metadata/schemas/data-source.json

```diff
@@ -0,0 +1,49 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Data Source metadata",
+  "name": "data-source",
+  "namespace": "data-sources",
+  "properties": {
+    "schema_name": {
+      "type": "string",
+      "pattern": "^[a-z][a-z0-9-_]*[a-z0-9]$",
+      "minLength": 1
+    },
+    "display_name": {
+      "description": "The display name of the Data Source",
+      "type": "string",
+      "pattern": "^[a-zA-Z][a-zA-Z0-9-_. (){}]*[a-zA-Z0-9)}]$"
+    },
+    "metadata": {
+      "description": "Additional data specific to this Data Source",
+      "type": "object",
+      "properties": {
+        "description": {
+          "description": "Data source description",
+          "type": "string"
+        },
+        "source": {
+          "description": "URL to access data source",
+          "type": "string"
+        },
+        "accessKey": {
+          "description": "Access key to access data source",
+          "type": "string"
+        },
+        "accessSecret": {
+          "description": "Access secret key to access data source",
+          "type": "string"
+        },
+        "type": {
+          "description": "Format of data source (i.e. database, csv, etc.)",
+          "type": "string"
+        },
+        "code": {
+          "type": "array"
```

Comment:
I'm assuming this is a line of code per array element (like code-snippets).  We should add a description that reflects the field's usage.

We're missing [`language`](https://github.com/elyra-ai/elyra/pull/562/files#diff-400f837acca64f7a15e001f07288bc56R58) and it should probably be marked as required.